app/src/main/java/com/example/hackernews/data/CallApi.kt

```diff
@@ -3,93 +3,95 @@ package com.example.hackernews.data
 import android.content.Context
 import android.util.Log
 import android.widget.Toast
-import com.example.hackernews.comments.CommentsAdapter
+import com.example.hackernews.callbacks.LoadCommentCallback
+import com.example.hackernews.callbacks.LoadDataCallback
 import com.example.hackernews.constants.Api
 import com.example.hackernews.helpers.Helper
 import com.example.hackernews.models.Comment
 import com.example.hackernews.models.NewsM
-import com.example.hackernews.news.NewsAdapter
 import com.example.hackernews.news.NewsDataType
 import com.example.hackernews.services.NewsService
+import kotlinx.coroutines.*
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
 import retrofit2.Retrofit
 import retrofit2.converter.gson.GsonConverterFactory
 import java.io.Serializable
+import java.lang.Exception
 
 class CallApi(val context: Context) : Serializable{
 
-    private val retrofit: Retrofit by lazy {
+    private val retrofit: NewsService by lazy {
         Retrofit.Builder()
                 .baseUrl(Api.BASE_URL)
                 .addConverterFactory(GsonConverterFactory.create())
-                .build()
+                .build().create(NewsService::class.java)
     }
+    private var job: Job? = null
 
-    private val newsApi by lazy {
-        retrofit.create(NewsService::class.java)
-    }
-
-    fun getStories(newsDataType: NewsDataType, newsAdapter: NewsAdapter) {
-        val storiesIds = ArrayList<Int>()
-        val call = newsApi.getStoriesIds(newsDataType.rawValue).enqueue(object : Callback<List<Int>> {
-            override fun onResponse(call: Call<List<Int>>, response: Response<List<Int>>) {
-                if (response.isSuccessful) {
-                    for (i in response.body()!!.indices) {
-                        storiesIds.add(response.body()!![i])
-                        loadNews(response.body()!![i], newsAdapter)
+    fun getStories(newsDataType: NewsDataType, loadData: LoadDataCallback) {
+        val allNews = ArrayList<NewsM>()
+        job = CoroutineScope(Dispatchers.IO).launch {
+            val storiesIds = retrofit.getStoriesIds(newsDataType.rawValue).body() as ArrayList<Int>
+            (storiesIds.indices).forEach {
```

Comment:
You need elements of `storiesId` array in that case use only `forEach` 

```kotlin 
storiesId.forEach { storyId ->
   val news = loadNews(storyId)
   ...
}
```

But if you need index and element there is a function called `forEachIndexed { index, element -> ... }`